compiler/noirc_evaluator/src/ssa/ssa_gen/mod.rs

```diff
@@ -232,10 +233,26 @@ impl<'a> FunctionContext<'a> {
                 Ok(self.builder.numeric_constant(*value as u128, Type::bool()).into())
             }
             ast::Literal::Str(string) => Ok(self.codegen_string(string)),
-            ast::Literal::FmtStr(string, number_of_fields, fields) => {
+            ast::Literal::FmtStr(fragments, number_of_fields, fields) => {
+                let mut string = String::new();
+                for fragment in fragments {
+                    match fragment {
+                        FmtStrFragment::String(value) => {
+                            // Escape curly braces in non-interpolations
+                            let value = value.replace('{', "{{").replace('}', "}}");
```

Comment:
I'm realizing that for codegen purposes we could just escape `{` as `\{`, though maybe for consistency with everything else it's fine if we use the same approach as the Noir syntax.